### ISSUE TITLE
Static IP override via USB but fallback to DHCP if no connectivity

### ISSUE THREAD

**Issue by mihaime** [lf-edge/eve#3990]
#### Describe the unexpected behaviour

I'm staging a server with a custom config of LF-Edge and an USB stick that has override for static IP on eth0.
I plugged a laptop into eth0 just for doing ping tests to confirm that the IP got configured.
Ping works for 10-15 seconds after which it fails.

If I look in the logs I see
"Not testing highest priority DevicePortConfig key usb due to All attempts to connect to controller...../edgedevice/ping failed: (send vai eth0 with SRC IP)"

after that the IP on eth0 goes away and it switched to DHCP 
ERROR: Summary: Waiting for DHCP IP address(es)

It will only switch back to having an IP on eth0 (from the override) if I do:
- ifconfig down/up
- or a reboot

#### Expected behavior

That the static IP override remains in place.

Is this how it should behave by design with fallback to DHCP and ignoring the static IP override on eth0 if it cannot reach controller?

**Comment by rucoder** [comment]
yes, this is by design. If the controller is available then this fallback won't happen

**Comment by mihaime** [comment]
@rucoder but then to go back to the static IP override I need to physically unplug the network cable or reboot the device?

I think on a logic level if someone sets a static IP, the reason must be he does not have DHCP in that network.
If you fallback to DHCP and never come back to the static IP override, then that ends up in a stuck in time moment.
I don't see how this fallback and remaining in it can be helping to get out of the loss of connectivity case.

**Comment by rucoder** [comment]
For eve it doesn't matter how to get IP as soon as cloud is reachable. If your network  has dhcp but the cloud cannot be reached it sgould try static again, but if the cloud is reachnable over dhcp then no. To test it properly setup your dhcp to provide invalid ip

**Comment by mihaime** [comment]
Thank you @rucoder. That clears up my misunderstanding and sorry for it!

Seems like the logic is built well into eve!

Will test in my lab!
